### ISSUE TITLE
Spies don't notice there's no technology to steal

### ISSUE THREAD

**Issue by axatin** [LoneGazebo/Community-Patch-DLL#8605]
_1. Mod version (X.Y.Z, e.g. 1.3.4):_
1.3.4

_2. Mod list (if using Vox Populi only, leave blank):_
Community Patch only

_3. Error description:_ 
Spies sometimes continue "gathering information" even though there's no tech available to steal. When they are finished, an empty list to select from appears, and it's not possible to continue the game from this point on because there's no way to leave that view.

The error seems to appear only when the spy successfully stole a tech before. Maybe a check whether a stolen technology was the last one available is missing... The error wasn't there in previous mod versions, I think.

![Screenshot (24)](https://user-images.githubusercontent.com/95587882/152439628-70197b85-2cdf-4272-bf88-435c33edd5a8.png)

**Comment by RecursiveVision** [comment]
Might be fixed in latest version (1.4, just released now), reopen if not fixed.

**Comment by axatin** [comment]
Unfortunately, the error still occurs in the new version.

**Comment by RecursiveVision** [comment]
@LoneGazebo

**Comment by axatin** [comment]
ok I just figured it out and it might have been my own fault. I play Community Patch only and don't want to have any other balance/gameplay changes, including the larger number of spies on large maps. In the mod XMLs I saw a file "CommunityPatchDLLChanges.xml" which looked to me like a file to customize mod options, and I manually set "BALANCE_CORE_SPIES" to zero. And that was a problem because then, in the following code, the if-clause in line 504 never evaluates to true.
![Screenshot (41)](https://user-images.githubusercontent.com/95587882/154727232-d890514d-3145-499e-8e6d-1a6d4bcfbfa9.png)
I assume I wasn't supposed to just change that, and I'm really sorry if I wasted any of your time.
